bin/fixmystreet.com/call-wasteworks-backend

```diff
@@ -37,10 +38,42 @@ my $cfg = $cobrand->feature($opts->backend);
 my $class = $opts->backend eq 'echo' ? 'Integrations::Echo' : 'Integrations::Bartec';
 my $integration = $class->new(%$cfg);
 
+my $lockfile;
+if ($opts->fork) {
+    unless ($opts->out) {
+        die "Must use --out with --fork";
+    }
+
+    $lockfile = $opts->out . ".lock";
+
+    if ( -e $opts->out ) {
+        # file already exists, must have been put there by fork()ed previous
+        # incantation of this script.
+
+        unlink $lockfile if -e $lockfile; # tidy up lock file just in case
+        exit 0;
+    }
+
+    if ( -e $lockfile ) {
+        # don't fork, there's already a forked process working on this
+        # batch of API calls
+        exit 1;
+    }
+    my $pid = fork();
+    if ($pid) {
+        # we're the parent, let the caller know that the data isn't ready yet
+        exit 1;
+    } else {
+        # create the lockfile so we don't fork() the same batch of jobs again
+        open(FH, ">", $lockfile) and close(FH);
+    }
+}
+
 my $calls = decode_json($opts->calls);
```

Comment:
I think you need to trap this script dying and cleaning up the lock file then, just in case something happens.

Reply:
Is that a case of using an `END` block, or something lower level?

Reply:
`END` would work presumably, yeah, but my other comment about not changing this file and using flock instead sounds like a better way to go? Then there wouldn't be a lock file.

Reply:
Yep probably. Was trying to do the ~simplest changes first in case we needed to press this into production in the same form as it's been tested on staging the last couple of months. Don't think that's the case now though, so will look at flocking stuff.